chart/templates/_helpers.yaml

```
@@ -248,16 +248,30 @@ If release name contains chart name it will be used as a full name.
 
 {{/* This helper will change when customers deploy a new image */}}
 {{- define "airflow_image" -}}
-  {{- printf "%s:%s" (.Values.images.airflow.repository | default .Values.defaultAirflowRepository) (.Values.images.airflow.tag | default .Values.defaultAirflowTag) }}
+{{- $repository := .Values.images.airflow.repository | default .Values.defaultAirflowRepository -}}
+{{- $tag := .Values.images.airflow.tag | default .Values.defaultAirflowTag -}}
+{{- $digest := .Values.images.airflow.digest | default .Values.defaultAirflowDigest -}}
+{{- if $digest }}
+{{- printf "%s@%s" $repository $digest -}}
+{{- else }}
+{{- printf "%s:%s" $repository $tag -}}
+{{- end }}
```

```suggestion
  {{- $repository := .Values.images.airflow.repository | default .Values.defaultAirflowRepository -}}
  {{- $tag := .Values.images.airflow.tag | default .Values.defaultAirflowTag -}}
  {{- $digest := .Values.images.airflow.digest | default .Values.defaultAirflowDigest -}}
  {{- if $digest }}
    {{- printf "%s@%s" $repository $digest -}}
  {{- else }}
    {{- printf "%s:%s" $repository $tag -}}
  {{- end }}
```

We just had a few PRs to do some formatting cleanup, let's try and keep it 👍

---

maybe we should add pre commit to validate the needed format?

---

If we could that'd be awesome, but I'm not aware of anything we can use out of the box. Worth looking around a bit though.

---

followup issue https://github.com/apache/airflow/issues/30546